site/_posts/2021-03-08-establishing-reliability-as-code-via-sonobuoy.md

```diff
@@ -0,0 +1,112 @@
+livelinessProbe confirms the container is in a running state
+readinessProbe - confirms the container is ready to respond to requests 
+
+The reliability scanner checks allow Kubernetes cluster operators to report Pods that are missing the liveliness and readiness  probes, as part of the Sonobuoy report.  Currently only liveliness and readiness checks are recommended by the VMware CRE team, however this check could extend to include startupProbe, if necessary, for your clusters.  Additionally, a potential improvement in future releases would be to add a labeling capability for operators to specify which Pods to skip.
+
+### Owner annotations
+Owner annotations provide the ability to replicate patterns of deployment across multiple physical hosts. In a multi-tenant Kubernetes deployment, owner annotations support incident management by making it easy to figure out parties who should be notified.  The scanner will check any user-provided annotations and allowed domains, against the Services running in the cluster, to reports back if the desired values are not set appropriately.
+
+### Minimum desired quality of service
+Quality of service (QoS) in Kubernetes is used by the scheduler to make decisions around scheduling and evicting pods on a node. There are three classes of QoS in Kubernetes: **BestEffort**, **Burstable**, and **Guaranteed**.
+
+The upfront reservation on scheduled Nodes depend on how our application requirement was defined.  One of the reliability practices that allows applications to operate more efficiently is to ensure that the Node has or reserves enough resources when a container is scheduled.  The reliability scanner will report any Pod that does not meet the minimum desired QoS class defined in the constraint.
+
+### Getting Started
+With the Reliability Scanner, you can add new reliability checks easily.  Please refer to the [README](https://github.com/vmware-tanzu/sonobuoy-plugins/blob/master/reliability-scanner/README.md) for more details.
+
+First, make sure that [Sonobuoy CLI](https://github.com/vmware-tanzu/sonobuoy), Docker, make, ytt, and kubectl are installed on your local machine.  There’s a helpful [Makefile](https://github.com/vmware-tanzu/sonobuoy-plugins/blob/master/reliability-scanner/Makefile) that handles the necessary setup and templating needed to run the customized reliability check YAML.  
+
+
+### Adding a new QOS reliability check
+The reliability scanner is ready to be used to scan your cluster. However, a cluster operator who is interested in creating a customized reliability check, will need to make modifications in two places in order to create a new customized reliability check.  First, the new check must be defined by name, description, kind, and spec, in a [YAML configuration file](https://github.com/vmware-tanzu/sonobuoy-plugins/blob/master/reliability-scanner/plugin/reliability-scanner-custom-values.lib.yml) as shown below:
+
+```
+name: ”Pod QOS check”
```

Comment:
This looks like smart-quotes automatically generated by macOS, we should use plain quotes (straight vertical) since folks might copy-paste this.

Reply:
@wilsonehusin Should be all fixed now.  Thanks so much!